tests/chainer_tests/functions_tests/array_tests/test_rollaxis.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+
+
+@testing.parameterize(
+    {'axis': 0, 'start': 2, 'out_shape': (3, 2, 4)},
+    {'axis': 2, 'start': 0, 'out_shape': (4, 2, 3)},
+    {'axis': 1, 'start': 1, 'out_shape': (2, 3, 4)},
+    {'axis': -3, 'start': 2, 'out_shape': (3, 2, 4)},
+    {'axis': -1, 'start': 0, 'out_shape': (4, 2, 3)},
+    {'axis': -2, 'start': -2, 'out_shape': (2, 3, 4)},
```

Comment:
It is desirable to check all patterns to subtract or not subtract `ndim` from `axis` and/or `start`.

Specifically we should parameterize like this (sorry, I cannot come up with appropriate parameter names):
```
testring.parameterize_dict(
    [{'axis': 0, 'start': 2}, {'axis': 2, 'start': 0}, {'axis': 1, 'start': 1}],
    [{'axis_plus': 0, 'start_plus': 0}, {'axis_plus': 0, 'start_plus': -3}, {'axis_plus': -3, 'start_plus': 0}, {'axis_plus': -3 'start_plus': -3}])
```

We can also write second line as `testing.product({'axis_plus': [0, -3], 'start_plus': [0, -3]})`. But I think it incurs readability...